src/formatting/lists.rs

```diff
@@ -734,17 +739,31 @@ pub(crate) fn extract_post_comment(
     // Cleanup post-comment: strip separators and whitespace.
     let post_snippet = post_snippet[..comment_end].trim();
     let post_snippet_trimmed = if post_snippet.starts_with(|c| c == ',' || c == ':') {
-        post_snippet[1..].trim_matches(white_space)
+        post_snippet[1..].trim_matches(white_space).to_string()
     } else if let Some(post_snippet) = post_snippet.strip_prefix(separator) {
-        post_snippet.trim_matches(white_space)
+        post_snippet.trim_matches(white_space).to_string()
     }
     // not comment or over two lines
     else if post_snippet.ends_with(',')
         && (!post_snippet.trim().starts_with("//") || post_snippet.trim().contains('\n'))
     {
-        post_snippet[..(post_snippet.len() - 1)].trim_matches(white_space)
+        post_snippet[..(post_snippet.len() - 1)]
+            .trim_matches(white_space)
+            .to_string()
+    } else if post_snippet.find_uncommented(",").is_some() {
+        let sep_pos = post_snippet.find_uncommented(",").unwrap();
```

Comment:
While I'd love to avoid having to do this full scan more than once, we can certainly avoid having to do it a third time here by using an `if let Some..` in the conditional which will also be a bit more idiomatic

```suggestion
    } else if let Some(sep_pos) = post_snippet.find_uncommented(",") {
```